src/json/from_json.cr

```diff
@@ -266,9 +266,16 @@ end
 
 def NamedTuple.new(pull : JSON::PullParser)
   {% begin %}
-    {% for key in T.keys %}
-      %var{key.id} = nil
-      %found{key.id} = false
+    {% for key, type in T %}
+      {% if type.nilable? %}
+        %var{key.id} = nil
+      {% else %}
+        %var{key.id} = uninitialized typeof(begin
+          value = uninitialized self
+          value[{{ key.symbolize }}]
+        end)
```

Comment:
Wouldn't this work?
```suggestion
        %var{key.id} = uninitialized {{ type }}
```

Reply:
I think we should put a comment around why we do it this way. I think I made the exact same comment in a previous PR but now I can't remember why doing it the way @straight-shoota (and previously me) suggest is not a good idea.

Reply:
It's #11834. But there is an even more idiomatic way of expressing this: #12011